File: tests/perf/test-call-basic-1.js
/*
 *  Basic function call performance.
 */

if (typeof print !== 'function') { print = console.log; }

function test() {
    var i;

    function f() { return; }

    var t1 = Date.now();

    for (i = 0; i < 1e6; i++) {
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f();
    }

    var t2 = Date.now();
    print((1e6 * 100 / (t2 - t1)) + ' calls per millisecond');
}

try {
    test();
} catch (e) {
    print(e.stack || e);
    throw e;
}

File: tests/perf/test-call-basic-1.rb
def test()
    def f() return end

    i = 0
    while i < 1e6 do
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        i += 1
    end
end

test()
